Packages/com.unity.inputsystem/Tests/InputSystem/Plugins/AndroidSensorTests.cs

```diff
@@ -0,0 +1,75 @@
+#if UNITY_EDITOR || UNITY_ANDROID
+using UnityEngine;
+using UnityEngine.Experimental.Input;
+using UnityEngine.Experimental.Input.Plugins.Android;
+using UnityEngine.Experimental.Input.Plugins.Android.LowLevel;
+using NUnit.Framework;
+
+class AndroidSensorTests : InputTestFixture
+{
+    [Test]
+    [Category("Devices")]
+    public void Devices_CanCreateSensors()
+    {
+        var sensorTypes = new[] {
+            typeof(AndroidAccelerometer),
+            typeof(AndroidMagneticField),
+            typeof(AndroidOrientation),
+            typeof(AndroidGyroscope),
+            typeof(AndroidLight),
+            typeof(AndroidPressure),
+            typeof(AndroidProximity),
+            typeof(AndroidTemperature),
+            typeof(AndroidGravity),
+            typeof(AndroidLinearAcceleration),
+            typeof(AndroidRotationVector),
+            typeof(AndroidRelativeHumidity),
+            typeof(AndroidAmbientTemperature),
+            typeof(AndroidMagneticFieldUncalibrated),
+            typeof(AndroidGameRotationVector),
+            typeof(AndroidGyroscopeUncalibrated),
+            typeof(AndroidSignificantMotion),
+            typeof(AndroidStepDetector),
+            typeof(AndroidStepCounter),
+            typeof(AndroidGeomagneticRotationVector),
+            typeof(AndroidHeartRate)
+        };
+
+        foreach (var s in sensorTypes)
+        {
+            var setup = new InputDeviceBuilder(s.Name);
+            var device = setup.Finish();
+
+            Assert.That(device, Is.AssignableTo<Sensor>());
+        }
+    }
+
+    private InputDeviceDescription GetSensorDescription(AndroidSenorType type)
+    {
+        return new InputDeviceDescription
+        {
+            interfaceName = "Android",
+            deviceClass = "AndroidSensor",
+            capabilities = new AndroidSensorCapabilities()
+            {
+                sensorType = type
+            }.ToJson()
+        };
+    }
+
+    [Test]
+    [Category("Devices")]
+    public void Devices_AccelerometerWorks()
```

Comment:
Avoid "works"-style tests. Have tests specify more or less exactly what they're after.

Reply:
(my own "SupportsX" kinda straddle that border...)